contrib/haproxy/stream-processing-offload/haproxy_messages.go

```diff
@@ -0,0 +1,145 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025 Datadog, Inc.
+
+package streamprocessingoffload
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/DataDog/dd-trace-go/v2/instrumentation/appsec/proxy"
+	"github.com/negasus/haproxy-spoe-go/message"
+	"github.com/negasus/haproxy-spoe-go/request"
+
+	"github.com/DataDog/dd-trace-go/v2/ddtrace/ext"
+	"github.com/DataDog/dd-trace-go/v2/ddtrace/tracer"
+)
+
+var _ proxy.RequestHeaders = (*messageRequestHeaders)(nil)
+var _ proxy.ResponseHeaders = (*responseHeadersHAProxy)(nil)
+var _ proxy.HTTPBody = (*messageBody)(nil)
+var _ proxy.HTTPBody = (*messageBody)(nil)
+
+type messageRequestHeaders struct {
+	req     *request.Request
+	msg     *message.Message
+	hasBody bool
+}
+
+func (m *messageRequestHeaders) ExtractRequest(_ context.Context) (proxy.PseudoRequest, error) {
+	headers, err := parseHAProxyReqHdrsBin(getBytesArrayValue(m.msg, "headers"))
+	if err != nil {
+		return proxy.PseudoRequest{}, err
+	}
+
+	authority := headers.Get("Host")
+	method := getStringValue(m.msg, "method")
+	path := getStringValue(m.msg, "path")
+	https := getBoolValue(m.msg, "https")
+
+	if authority == "" || method == "" || path == "" {
+		return proxy.PseudoRequest{}, fmt.Errorf("missing required values in the http request SPOE message")
+	}
+
+	scheme := "http"
+	if https {
+		scheme = "https"
+	}
+
+	// Define if a body is present, based on Content-Length header
+	contentLength := headers.Get("Content-Length")
+	if contentLength != "" {
```

Comment:
`contentLength` is only used inside the if statement
```suggestion
	if contentLength := headers.Get("Content-Length"); contentLength != "" {
```